src/main/webapp/resources/js/components/samples/components/fastqc/FastQC.jsx

```diff
@@ -0,0 +1,95 @@
+import { Button, Modal, Space, Typography } from "antd";
+import React from "react";
+import { FastQCMenu } from "./FastQCMenu";
+
+const { Text } = Typography;
+import { IconArrowLeft } from "../../../icons/Icons";
+import { clearFastQCData, setFastQCDetails } from "./fastQCSlice";
+import { useDispatch, useSelector } from "react-redux";
+import { getFastQCDetails } from "../../../../apis/files/sequence-files";
+import { InfoAlert } from "../../../alerts";
+
+/**
+ * Function to render fastqc modal
+ * @returns {JSX.Element}
+ * @constructor
+ */
+export function FastQC() {
+  const dispatch = useDispatch();
+
+  const qcEntryTranslations = {
+    UNPROCESSED: i18n("FastQC.sequencingobject.UNPROCESSED"),
+    QUEUED: i18n("FastQC.sequencingobject.QUEUED"),
+    ERRORED: i18n("FastQC.sequencingobject.ERRORED"),
+    PROCESSING: i18n("FastQC.sequencingobject.PROCESSING"),
+  };
+
+  const {
+    fileLabel,
+    fastQCModalVisible,
+    sequencingObjectId,
+    fileId,
+    fastQC,
+    processingState,
+  } = useSelector((state) => state.fastQCReducer);
+
+  React.useEffect(() => {
+    getFastQCDetails(sequencingObjectId, fileId).then(
+      ({ analysisFastQC, sequenceFile, sequencingObject }) => {
+        dispatch(
+          setFastQCDetails({
+            fastQC: analysisFastQC,
+            file: sequenceFile,
+            processingState: sequencingObject.processingState,
+          })
+        );
+      }
+    );
+  }, [sequencingObjectId, fileId]);
+
+  return (
+    <>
```

Comment:
Don't need the empty fragment, just start with `return fastQCModalVisible ? ....`, and since visible is only used once in this file why not just call it visible?  And since you are not lazy loading the modal, you actualy don't need to wrap the modal in the `fastQCModalVisible` at all, the modal will handle it all for you since you have that on line 74.

Reply:
The reason it is called fastQCModalVisible is because it is a value stored in the state. This way we know exactly what that is for